test/Microsoft.Azure.ServiceBus.UnitTests/TransactionTests.cs

```diff
@@ -240,6 +240,10 @@ public class TransactionTests
                     ts.Dispose();
                 }
 
+                // Adding delay since transaction Commit/Rollback is an asynchronous operation.
+                // Operating on the same message should not be done.
+                await Task.Delay(TimeSpan.FromSeconds(2));
```

Comment:
Similar fixes needed in
`TransactionalCompleteRollbackTest`
`TransactionRollbackWorksAcrossClientsUsingSameConnectionToSameEntity`

Reply:
Added those. Might have gone a bit extra - please review 🙂